src/rmind/utils/functional.py

```diff
@@ -37,3 +37,21 @@ def diff_last(input: Tensor, n: int = 1, *, append: float | None = None) -> Tens
         else None
     )
     return torch.diff(input, n=n, dim=-1, append=append_)
+
+
+class SignalWithThresholdResult(NamedTuple):
+    class_idx: Tensor
+    prob: Tensor
+
+
+def non_zero_signal_with_threshold(
+    logits: Tensor, threshold: float = 0.85
```

Comment:
i was thinking about where to put this thr in config. so far its just a magic number here

Reply:
makes sense. since it categorical we can compute precision / recall over val set and use that as a threshold